### ISSUE TITLE
Incorrect user filtering BY NAME: in availability condition UI

### ISSUE THREAD

**Issue by semteacher** [Wunderbyte-GmbH/moodle-mod_booking#456]
The autocomplete field uder availability condition UI returns incorrect list of users when being filtered BY NAME:

![Image](https://github.com/Wunderbyte-GmbH/moodle-mod_booking/assets/1571998/7c9a2b52-0f66-45c1-9996-eff45709b3ce)

![Image](https://github.com/Wunderbyte-GmbH/moodle-mod_booking/assets/1571998/68070374-f486-4a4b-8b05-5e38159a2791)

This always happens unde MySQL and most time (but not every time!) under Postgress. 
As result, the following behat tests fails:
```
Configure combined availability conditions - overbooking given to user
Configure user-dependent availability condition
```
Caused by load_users() method

**Comment by semteacher** [comment]
The following loop in the code of load_users() method 

![Image](https://github.com/Wunderbyte-GmbH/moodle-mod_booking/assets/1571998/d5f236f4-7240-460c-8422-1a7887814874)
is definitely responsible for above issue 

![Image](https://github.com/Wunderbyte-GmbH/moodle-mod_booking/assets/1571998/e28b6eb3-bd99-4b53-8c10-139413d91886)

I would recommend to simplify search to just single ILIKE

**Comment by bernhard-wunderbyte** [comment]
Should be fixed with commit 6d004d2c454a70947140ed81d1a8a85efedd2c21
